packages/calypso-products/src/get-jetpack-product-or-plan-display-name.ts

```diff
@@ -0,0 +1,15 @@
+import { getJetpackProductDisplayName } from './get-jetpack-product-display-name';
+import { getPlan } from './main';
+import type { Product } from './types';
+import type { TranslateResult } from 'i18n-calypso';
+
+/**
+ * Get Jetpack product display name based on the product purchase object.
+ */
+export function getJetpackProductOrPlanDisplayName(
+	product: Product
+): TranslateResult | undefined {
+	return product?.product_name
```

Comment:
```suggestion
	return product.product_name
```

product cannot be `undefined` or `null` at this point